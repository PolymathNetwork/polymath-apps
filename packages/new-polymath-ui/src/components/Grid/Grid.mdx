---
name: Grid
menu: Primitives
---

import { Playground, PropsTable } from 'docz';
import { Grid, GridDocz } from './';

# Grid

<PropsTable of={GridDocz} />

## Basic usage

<Playground>
  <Grid gridAutoFlow="column" gridAutoColumns="1fr" alignItems="end">
    <div>First block</div>
    <div>Second block</div>
  </Grid>
</Playground>

## 12 columns Grid system

<Playground>
  <Grid.Row>
    <Grid.Col gridSpan={[12, 12, 12, 5]}>First block</Grid.Col>
    <Grid.Col gridSpan={[12, 12, 12, 5]}>Second block</Grid.Col>
  </Grid.Row>
</Playground>
