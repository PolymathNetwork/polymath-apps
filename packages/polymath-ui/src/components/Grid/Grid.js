// @flow

import styled from 'styled-components';
import {
  gridGap,
  gridAutoFlow,
  gridAutoColumns,
  alignItems,
  justifyContent,
} from 'styled-system';

import Box from '../Box';

const Grid = styled(Box)`
  display: grid;
  ${gridGap};
  ${gridAutoFlow};
  ${gridAutoColumns};
  ${alignItems};
  ${justifyContent};
`;

Grid.defaultProps = {
  gridGap: 20,
};

export default Grid;
