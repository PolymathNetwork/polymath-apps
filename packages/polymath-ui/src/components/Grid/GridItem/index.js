import React from 'react';
import styled from 'styled-components';
import { gridColumn, gridRow, gridArea } from 'styled-system';

const GridItem = styled.div`
  ${gridColumn};
  ${gridRow};
  ${gridArea};
`;

GridItem.defaultProps = {};

export default GridItem;
