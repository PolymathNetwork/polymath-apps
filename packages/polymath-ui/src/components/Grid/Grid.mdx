---
name: Grid
menu: Primitives
---

import { Playground, PropsTable } from 'docz';
import Grid from './';

# Grid

<PropsTable of={Grid} />

## Basic usage

<Playground>
  <Grid gridAutoFlow="column" gridAutoColumns="1fr" alignItems="end">
    <div>First block</div>
    <div>Second block</div>
  </Grid>
</Playground>
