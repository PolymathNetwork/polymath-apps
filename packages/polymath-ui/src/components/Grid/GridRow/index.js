import styled from 'styled-components';

import Grid from '../Grid';

const GridRow = styled(Grid)`
  grid-template-columns: repeat(12, 1fr);
`;

export default GridRow;
