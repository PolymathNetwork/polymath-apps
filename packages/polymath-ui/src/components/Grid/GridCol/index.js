import styled from 'styled-components';
import { style } from 'styled-system';

import GridItem from '../GridItem';

const gridSpan = style({
  prop: 'gridSpan',
  cssProperty: 'grid-column',
  transformValue: n => `auto / span ${n}`,
});

const GridCol = styled(GridItem)`
  ${gridSpan};
`;

GridCol.propTypes = {
  ...gridSpan.propTypes,
};

export default GridCol;
