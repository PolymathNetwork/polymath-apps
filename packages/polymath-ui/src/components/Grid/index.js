import Grid from './Grid';
import GridItem from './GridItem';
import GridRow from './GridRow';
import GridCol from './GridCol';

export default Object.assign(Grid, {
  Item: GridItem,
  Row: GridRow,
  Col: GridCol,
});
