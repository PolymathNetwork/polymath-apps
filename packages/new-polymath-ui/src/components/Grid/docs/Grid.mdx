---
name: Grid
menu: Primitives
---

import { Playground, PropsTable } from 'docz';
import { Grid } from '../';
import { GridRow } from '~/components/GridRow';

# Grid

<PropsTable of={Grid} />

## Basic usage

In its normal form, `Grid` accepts basic css-grid properties.

<Playground>
  <Grid gridAutoFlow="column" gridAutoColumns="1fr" alignItems="end">
    <div style={{ height: '200px', background: 'tomato' }}>First block</div>
    <div style={{ background: 'cyan' }}>Second block</div>
  </Grid>
</Playground>

## 12-columns Grid system

"Bootstrap-style" 12-columns grid is being used in the design. To achieve it you can use the `GridRow` component.
The main difference with the classic Bootstrap grid is that **`GridRow` is a global container for your grid, not only a row** (so, then the name is not really accurate but could not find better).

### Important

- — You can have defined any number of rows in a `GridRow`, just keep adding some `Grid.Col`.
- — `Grid.Col` and its `gridSpan` prop is just a helper for the native css-grid `grid-column` property.

<br />

<Playground>
  <GridRow>
    <GridRow.Col gridSpan={6} style={{ background: 'cyan' }}>
      Row 1 - First block
    </GridRow.Col>
    <GridRow.Col gridSpan={6} style={{ background: 'tomato' }}>
      Row 1 - Second block
    </GridRow.Col>
    <GridRow.Col gridSpan={6} style={{ background: 'cyan' }}>
      Row 2 - First block
    </GridRow.Col>
    <GridRow.Col gridSpan={6} style={{ background: 'tomato' }}>
      Row 2 - Second block
    </GridRow.Col>
  </GridRow>
</Playground>

You can see that the `grid-gap` is doing the job for you. No margin-top/bottom in the children. Your grid is consistent and your vertical flow is managed automatically.

## Responsive

As with every primitives, you can pass an array to the props to handle responsive (try resizing your browser):

<Playground>
  <GridRow>
    <GridRow.Col gridSpan={[12, 12, 7, 6]} style={{ background: 'cyan' }}>
      Row 1 - First block
    </GridRow.Col>
    <GridRow.Col gridSpan={[12, 12, 5, 6]} style={{ background: 'tomato' }}>
      Row 1 - Second block
    </GridRow.Col>
    <GridRow.Col gridSpan={[12, 12, 12, 6]} style={{ background: 'cyan' }}>
      Row 2 - First block
    </GridRow.Col>
    <GridRow.Col gridSpan={[12, 12, 12, 6]} style={{ background: 'tomato' }}>
      Row 2 - Second block
    </GridRow.Col>
  </GridRow>
</Playground>

## Span

To span (leave an empty space) between two columns, you use the native `grid-column` css property:

<Playground>
  <GridRow>
    <GridRow.Col gridSpan={6} style={{ background: 'cyan' }}>
      Row 1 - First block
    </GridRow.Col>
    <GridRow.Col gridColumn="8 / span 5" style={{ background: 'tomato' }}>
      Row 1 - Second block
    </GridRow.Col>
  </GridRow>
</Playground>
