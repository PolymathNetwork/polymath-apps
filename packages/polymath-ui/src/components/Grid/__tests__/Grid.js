import React from 'react';
import renderer from 'react-test-renderer';

import Grid from '../';

test('renders without crashing', () => {
  const component = renderer.create(
    <Grid gridAutoFlow="column" gridAutoColumns="1fr" alignItems="end">
      <div>First block</div>
      <div>Second block</div>
    </Grid>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
