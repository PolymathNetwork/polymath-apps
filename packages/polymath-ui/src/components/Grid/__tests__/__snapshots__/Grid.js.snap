// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c0 {
  box-sizing: border-box;
  display: grid;
  grid-gap: 20px;
  grid-auto-flow: column;
  grid-auto-columns: 1fr;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
}

<div
  className="c0"
>
  <div>
    First block
  </div>
  <div>
    Second block
  </div>
</div>
`;
